component/core/src/main/java/org/exoplatform/social/core/jpa/storage/dao/jpa/SpaceDAOImpl.java

```diff
@@ -88,4 +88,50 @@ public SpaceEntity getSpaceByPrettyName(String spacePrettyName) {
     }
   }
 
+  @Override
+  public List<SpaceEntity> getCommonSpaces(String userId, String profileUserId, int offset, int limit) {
+    if (userId == null || userId == "") {
+      throw new IllegalArgumentException("userId is null or equals to 0");
+    }
+    if (profileUserId == null || profileUserId == "") {
+      throw new IllegalArgumentException("profileUserId is null or equals to 0");
+    }
+    if (offset < 0) {
+      throw new IllegalArgumentException("offset must be positive");
+    }
+    if (limit <= 0) {
+      throw new IllegalArgumentException("limit must be > 0");
+    }
+    TypedQuery<SpaceEntity> query = getEntityManager().createNamedQuery("SpaceEntity.getCommonSpacesBetweenTwoUsers",
+            SpaceEntity.class);
+    query.setParameter("user1", userId);
+    query.setParameter("user2", profileUserId);
+    query.setFirstResult(offset);
+    query.setMaxResults(limit);
+    return query.getResultList();
+
+  }
+
+  @Override
+  public int countCommonSpaces(String userId, String profileUserId) {
+    if (userId == null || userId == "") {
+      throw new IllegalArgumentException("userId is null or equals to 0");
+    }
+    if (profileUserId == null || profileUserId == "") {
+      throw new IllegalArgumentException("profileUserId is null or equals to 0");
+    }
+
+    Query q = getEntityManager().createNativeQuery("SELECT count(*) FROM SOC_SPACES t where t.SPACE_ID in " +
+            "( SELECT distinct t1.SPACE_ID FROM SOC_SPACES_MEMBERS t1  " +
+            "where t1.USER_ID = :user1  " +
+            "and (SELECT distinct t2.SPACE_ID FROM SOC_SPACES_MEMBERS t2 " +
+            "where t1.SPACE_ID=t2.SPACE_ID and t2.USER_ID = :user2 ))");
+    q.setParameter("user1", userId);
+    q.setParameter("user2", profileUserId);
+    try {
+      return ((Number) q.getSingleResult()).intValue();
+    } catch (NoResultException e) {
```

Comment:
Not sure the try catch is necessary.
Can you make a test with 2 users which have not common spaces ? 
The count should return 0 nativelly

Reply:
yep exactly, it returns 0 so it's not necessary